### ISSUE TITLE
[BUG] Not filling in all translations causes 404s and exceptions

### ISSUE THREAD

**Issue by jnvsor** [AnimalDesign/bolt-translate#82]
Simply creating a page and marking it as published in a default bolt install with translate goes very wrong.

Assuming the defaults, the English version throws an exception, and the german version throws a 404 (Despite showing up in the default "All pages" list)
## Details
- Translate Version: 4
- Install type: Extension Repository
- Bolt Version: 3.1.4
- PHP version: both 5.6 and 7
- Used webserver: Apache 2.4.23-5
## Reproduction
- Plain bolt install:
  `composer create-project bolt/composer-install <root> --prefer-dist`
- Go to website, make user, login.
- Go to extensions and add translate.
- Edit page in `contenttypes.yml` to include `dedata`, `endata`, `locale` etc fields and add `is_translateable: true` to title field as per readme
- Create a published page
  - title: test
  - status: published

Despite showing up in the german "All pages" list, the german version 404s

Viewing the page in English throws:

```
Symfony\Component\Routing\Exception\InvalidParameterException thrown with message "Parameter "slug" for route "contentlink" must match "[^/]++" ("" given) to generate a corresponding URL."

Stacktrace:
#17 Symfony\Component\Routing\Exception\InvalidParameterException in <root>/vendor/symfony/routing/Generator/UrlGenerator.php:177
#16 Symfony\Component\Routing\Generator\UrlGenerator:doGenerate in <root>/vendor/symfony/routing/Generator/UrlGenerator.php:136
#15 Symfony\Component\Routing\Generator\UrlGenerator:generate in <root>/vendor/bolt/bolt/src/Routing/UrlGeneratorFragmentWrapper.php:42
#14 Bolt\Routing\UrlGeneratorFragmentWrapper:generate in <root>/extensions/vendor/animal/translate/src/Routing/LocalizedUrlGenerator.php:42
#13 Bolt\Extension\Animal\Translate\Routing\LocalizedUrlGenerator:generate in <root>/extensions/vendor/animal/translate/src/TranslateExtension.php:254
#12 Bolt\Extension\Animal\Translate\TranslateExtension:getCurrentLocaleStructure in <root>/extensions/vendor/animal/translate/src/TranslateExtension.php:208
#11 Bolt\Extension\Animal\Translate\TranslateExtension:Bolt\Extension\Animal\Translate\{closure} in <root>/vendor/pimple/pimple/lib/Pimple.php:201
#10 Pimple:{closure} in <root>/vendor/pimple/pimple/lib/Pimple.php:83
#9 Pimple:offsetGet in <root>/vendor/bolt/bolt/src/Render.php:57
#8 Bolt\Render:render in <root>/vendor/bolt/bolt/src/Controller/Base.php:76
#7 Bolt\Controller\Base:render in <root>/vendor/bolt/bolt/src/Controller/Frontend.php:182
#6 Bolt\Controller\Frontend:record in <root>/extensions/vendor/animal/translate/src/Frontend/LocalizedFrontend.php:78
#5 Bolt\Extension\Animal\Translate\Frontend\LocalizedFrontend:record in <root>/vendor/symfony/http-kernel/HttpKernel.php:144
#4 call_user_func_array in <root>/vendor/symfony/http-kernel/HttpKernel.php:144
#3 Symfony\Component\HttpKernel\HttpKernel:handleRaw in <root>/vendor/symfony/http-kernel/HttpKernel.php:64
#2 Symfony\Component\HttpKernel\HttpKernel:handle in <root>/vendor/silex/silex/src/Silex/Application.php:586
#1 Silex\Application:handle in <root>/vendor/silex/silex/src/Silex/Application.php:563
#0 Silex\Application:run in <root>/public/index.php:14
```

CSV export of database row:

```
id,slug,datecreated,datechanged,datepublish,datedepublish,username,ownerid,status,templatefields,title,image,teaser,body,template,locale,dedata,deslug,endata,enslug
1,test,2016-10-09 20:03:29,2016-10-09 20:03:29,2016-10-09 20:03:04,,"",1,published,[],test,[],"","","","","","",[],test
```

Switching to the german version and saving there too fixes the problem

**Comment by frozenshadow** [comment]
@jnvsor this is a known problem (see issue #70).
A workaround (although not optimal) for now is to save all the possible translations, until @SahAssar fixes this ;)

**Comment by madc** [comment]
I think, this is the same issue as seen in #92.

**Comment by SahAssar** [comment]
Should have been fixed in #96, which fixed #92, so I'm closing. Please reopen if it reappears!
